### ISSUE TITLE
Bad font rendering in alacritty

### ISSUE THREAD

**Issue by dase78** [gokcehan/lf#1532]
I have an issue with lf in alacritty: the rendering of font/bar is not doing well.

I first thought, this is an Alacritty issue but after some testing it seems to me, as it was lf because in other TUIs (cmus, micro), everything is fine. This does not happen in Lxterminal or others, just with lf in combination to Alacritty.

Left pane Lxterminal, right pane Alacritty:
![scrot_20231210_084231](https://github.com/gokcehan/lf/assets/100223979/b2a8b159-6c8b-4a65-ab7a-2262ab64f214)
![scrot_20231210_090157](https://github.com/gokcehan/lf/assets/100223979/6ab422f7-23b5-474f-b508-9ac5b62fc994)

I use Hack Nerd Font Mono. Any ideas?

I forgot: Debian Stable (12), Alacritty 0.11.0, LF 28 (no Build-Environment, Stable Version from official repo)

**Comment by dase78** [comment]
Yes, Liberation Mono, generic monospace. It always is this kind of blurish.

**Comment by og900aero** [comment]
> Yes, Liberation Mono, generic monospace. It always is this kind of blurish. EDIT: In combination with Alacritty.

Please paste your alacritty conf file (alacritty.yml) without comments.

**Comment by dase78** [comment]
Yes, the comments...

`env:
    TERM:       xterm-256color

window:
  padding:
    x:          3
    y:          3
  decorations:  none
  opacity:      0

font:
  normal:
    family:     Hack Nerd Font Mono
    style:      Regular
  bold:
    family:     Hack Nerd Font Mono
    style:      Regular
  italic:
    family:     Hack Nerd Font Mono
    style:      Regular
  bold_italic:
    family:     Hack Nerd Font Mono
    style:      Regular
  size:         10

draw_bold_text_with_bright_colors: true

colors:
  primary:
    background: '0x2e3436'
    foreground: '0xd3d7cf'
  normal:
    black:      '0x2e3436'
    red:        '0xcc0000'
    green:      '#82a612'
    yellow:     '0xc4a000'
    blue:       '0x3465a4'
    magenta:    '0x75507b'
    cyan:       '0x06989a'
    white:      '0xd3d7cf'
  bright:
    black:      '0x555753'
    red:        '0xef2929'
    green:      '#a5c33f'
    yellow:     '0xfce94f'
    blue:       '0x729fcf'
    magenta:    '0xad7fa8'
    cyan:       '0x34e2e2'
    white:      '0xeeeeec'

selection:
  save_to_clipboard: true

cursor:
  style:
    shape:      Block
    blinking:   Always
  blink_interval: 600
  blink_timeout: 0

mouse:
  hide_when_typing: true
`

**Comment by og900aero** [comment]
> Yes, the comments...
> 
Rename the alacritty.yml file to alacritty.backup, then restart alacritty and try now.

**Comment by dase78** [comment]
Yes, it looks a bit "better", but of course not, how I want it to be... Perhaps I will just stick with lxterminal.
![scrot_20231213_120323](https://github.com/gokcehan/lf/assets/100223979/49a76ca5-53bb-4fda-b3b7-d58371816b63)

**Comment by og900aero** [comment]
> Yes, it looks a bit "better", but of course not, how I want it to be... Perhaps I will just stick with lxterminal. ![scrot_20231213_120323](https://private-user-images.githubusercontent.com/100223979/290167701-49a76ca5-53bb-4fda-b3b7-d58371816b63.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI0NjYzMDMsIm5iZiI6MTcwMjQ2NjAwMywicGF0aCI6Ii8xMDAyMjM5NzkvMjkwMTY3NzAxLTQ5YTc2Y2E1LTUzYmItNGZkYS1iM2I3LWQ1ODM3MTgxNmI2My5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBSVdOSllBWDRDU1ZFSDUzQSUyRjIwMjMxMjEzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDIzMTIxM1QxMTEzMjNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lNWE0OTM2ZTA3N2Q5YmIwYmFiMTM5OWE1ZGNlNGFmMzU3NzRjZWJkODM4MzliYWE0OWEyNjc4ODU3OTAyNzQ0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.xCk70mA6Q_wuHRJR_tl9m5DMczRH0zYjbJNTMH51tsY)

So then I can only think that the video card and the driver it can cause this problem, because Alacritty uses GPU acceleration. Alacritty, for example, works well and very fast for me, but the wezterm terminal is very slow, even though it also uses GPU acceleration... That's why I sometimes use urxvt and sometimes st terminal.

**Comment by dase78** [comment]
Yes, I will stick with Lxterminal. But it's a strange issue, because it's only this special combination: LF + Alacritty.

**Comment by saygo-png** [comment]
@dase78  The font looks the same to me in lf and micro in Alacritty, only the color of selected text is different. Its gray in alacritty making it look more blurry. The font does look different between terminals because the left terminal has turned on subpixel rendering. Try and see if you can enable it in alacritty  too or in debians font config util and change the selection color to make the font rendering the same between terminals

**Comment by dase78** [comment]
Thank you saygo-png. I used to play around with the font config via dpkg-reconfigure fontconfig... It did not help. I don't know what sort of jam my problem is, but I gave up on it - and got happy (again) with my unspectacular Lxterminal.

**Comment by joelim-work** [comment]
If it's the color that looks different, it's possibly because `tcell` thinks that your terminal doesn't [support 24-bit color](https://github.com/gdamore/tcell?tab=readme-ov-file#24-bit-color), so it tries to map them to a reduced palette (e.g. 256 color) when drawing the UI. Possibly related to #1743

**Comment by dase78** [comment]
No, it's not (it wasn't) the color, it's this blurry font in Alacritty, see screenshots in first post.

But I am not into this problem anymore as I just went back to Lxterminal (as mentioned).

I just let this open for anyone else to reply if same issues.

But thanks for response.

**Comment by CatsDeservePets** [comment]
I will close this issue due to keeping the issue tracker clean. If users encounter something related, they are still able to comment below. If there is any need for it, I will re-open this issue.

**Comment by DusanLesan** [comment]
> keeping the issue tracker clean

@CatsDeservePets Thank you for cleaning up the issues during the last period. It looks much better now without stale and unreproducible issues.

**Comment by CatsDeservePets** [comment]
@DusanLesan 

I know right! It makes it much simpler finding features/bugs to work on or users that might need help.

I'm trying to help wherever I can to make `lf` even better. :D
